include/quicr/quicr_server.h

```diff
@@ -165,7 +181,8 @@ class QuicRServer
    * API for unit test cases .
    */
   QuicRServer(std::shared_ptr<qtransport::ITransport> transport,
-              ServerDelegate& delegate, qtransport::LogHandler& logger);
+              ServerDelegate& delegate /* TODO: Considering shared or weak pointer */,
```

Comment:
+1 on passing shared_ptr instead of reference

Reply:
Same for logger

Reply:
I am fine with it being a reference .. this will ensure only the application has control on the delegate and not the library

Reply:
Added the comment to remind us later about it. Paul brought this up. There are others that might need to be updated... in a different PR.